### PR TITLE
Overview component creation

### PR DESCRIPTION
**Pull request by dwightdoran**
I've created the skeleton for the Overview widget. In doing so, I added some files and folders. Please look at everything and give honest feedback of file structure, naming conventions, etc. I'm happy to change anything that is an issue.

### CODE REVIEW

File: client/src/components/Overview/Overview.jsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import StyleSelector from './components/StyleSelector.jsx'
+import AddToCart from './components/AddToCart.jsx'
+import ImageGallery from './components/ImageGallery.jsx'
+
+import { getProductDataByItem } from '../../../../utils/apiCalls.js';
+
+
+class Overview extends React.Component {
+  constructor(props) {
+    super(props)
```
Comment (DominicMonares) [inline client/src/components/Overview/Overview.jsx]: semi-colon missing after super and state

File: client/src/components/Overview/Overview.jsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    return (
+      <div>
+        <div>
+          <h1>Overview Widget Here</h1>
+          <ProductInfo product_id={this.props.itemid}/>
+          <StyleSelector product_id={this.props.itemid} />
+          <AddToCart product_id={this.props.itemid} />
+          <ImageGallery product_id={this.props.itemid} />
+        </div>
+
```
Comment (DominicMonares) [inline client/src/components/Overview/Overview.jsx]: unnecessary empty line

File: client/src/components/Overview/components/AddToCart.jsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  }
+
+  render() {
+    return (
+      <div className="add-to-cart-info">Add To Cart Area</div>
+    )
+  }
+}
+
+export default AddToCart;
```
Comment (DominicMonares) [inline client/src/components/Overview/components/AddToCart.jsx]: newline required at end of doc

File: client/src/components/Overview/components/ImageGallery.jsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  }
+
+  render() {
+    return (
+      <div className="Image-Gallery">Image Gallery</div>
+    )
+  }
+}
+
+export default ImageGallery;
```
Comment (DominicMonares) [inline client/src/components/Overview/components/ImageGallery.jsx]: newline required at end of doc

File: client/src/components/Overview/components/ProductInfo.jsx [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  }
+
+  render() {
+    return (
+      <div className="product-info">Product Info</div>
+    )
+  }
+}
+
+export default ProductInfo;
```
Comment (DominicMonares) [inline client/src/components/Overview/components/ProductInfo.jsx]: newline required at end of doc

File: client/src/components/Overview/components/StyleSelector.jsx [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  }
+
+  render() {
+    return (
+      <div className="styles-info">Styles Info</div>
+    )
+  }
+}
+
+export default StyleSelector;
```
Comment (DominicMonares) [inline client/src/components/Overview/components/StyleSelector.jsx]: newline required at end of doc

File: client/src/components/Overview/components/AddToCart.jsx [hunk 7]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+
+
+class AddToCart extends React.Component {
+  constructor(props) {
+    super(props)
```
Comment (DominicMonares) [inline client/src/components/Overview/components/AddToCart.jsx]: semi-colon missing after super and state

File: client/src/components/Overview/components/ImageGallery.jsx [hunk 8]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+
+
+class ImageGallery extends React.Component {
+  constructor(props) {
+    super(props)
```
Comment (DominicMonares) [inline client/src/components/Overview/components/ImageGallery.jsx]: semi-colon missing after super and state

File: client/src/components/Overview/components/ProductInfo.jsx [hunk 9]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+
+
+class ProductInfo extends React.Component {
+  constructor(props) {
+    super(props)
```
Comment (DominicMonares) [inline client/src/components/Overview/components/ProductInfo.jsx]: semi-colon missing after super and state

File: client/src/components/Overview/components/StyleSelector.jsx [hunk 10]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+
+
+class StyleSelector extends React.Component {
+  constructor(props) {
+    super(props)
```
Comment (DominicMonares) [inline client/src/components/Overview/components/StyleSelector.jsx]: semi-colon missing after super and state

**Review by DominicMonares — CHANGES_REQUESTED** [review]
The logic looks solid! Only changes I notice are very minor syntax additions/changes, I made comments in each area. Looking good other than that.

Regarding the file structure, I can't remember if we decided on anything regarding child components for our widgets - whether we're putting them all in the same components folder or sub folders for each widget within that components folder. We should confirm with the rest of the group, but I like the way you have it structured and would vote to go in that direction.

**Comment by dwightdoran** [comment]
Changes made. Thanks for that @DominicMonares. Merging now

**Pull request merged**